modules/phase_field/src/userobjects/DiscreteNucleationInserter.C

```diff
@@ -78,15 +92,22 @@ DiscreteNucleationInserter::execute()
 
     const Real random = getRandomReal();
 
+    // branch the operation for using Poisson statistics (with time) or
+    // time-independent probability (e.g., recrystallization fraction)
+    if (!_poisson_stats)
+    {
+      if (getRandomReal() < rate)
```

Comment:
What is that?!

Reply:
what is what?

Reply:
What is that "non-poisson" statistics supposed to mean? It is completely ignoring element volume and time step. After implementing the carefully thought out poisson statistic this seems like a step back into the stone age. I would strongly advise against adding this. Unless I'm missing something big here...

Reply:
Sure, we should discuss this in more depth.  I attempted to describe the motivation for this in the description of the PR.  The point of this is exactly that, to ignore time.  Not quite sure I understand the bit about ignoring element volume - so maybe there needs to be a change there.  

The use case I'm envisioning is to look at subgrain formation leading to recrystallization.  In this situation, a dislocation network forms within a grain to form subgrains with some size population statistics.  As additional dislocations are added, the subgrain size distribution can change. The concept is of the subgrain size distribution and critical size is discussed in [Cram's paper, see Fig. 5](https://doi.org/10.1016/j.actamat.2009.07.024).
It is essentially instantaneous with respect to diffusional timescales of, e.g., grain boundary motion, so in my understanding, the probability should not include a time interval in which the zero event probability decreases with increasing time.  It's simply "at the end of this time step, there is this subgrain distribution, and some probability of those are critical size to become new grains.  Sample that and introduce nuclei."

Maybe I didn't use the right term by calling it non-Poisson or I am not thinking of the physics the right way.  Happy to change it to whatever is right.

Reply:
Let me clarify a little bit more.  In the situation I'm thinking of, there are two time scales with a very large separation.  The diffusive timescale of grain boundary motion and burnup accumulation is on the order of days to weeks, while the transformation of a subgrain into a stable nucleus by dislocation motion is at the speed of dislocation propagation.  Thus, the zero event probability of nucleation at the diffusive timescale for a given timestep is zero.  Every event that can happen will happen.

Reply:
Element size will be a factor though. You say *Every event that can happen will happen* but at most nQP events can happen per element. This may get saturated very quickly and result in the wrong statistics. My suggestion would be to write a completely separate inserter for that scenario that is not limited to QP positions but can randomly distribute a computed number of events in each element.

Reply:
okay, that's a good idea.